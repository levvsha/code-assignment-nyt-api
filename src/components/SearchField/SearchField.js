import './SearchField.styl';

import React from 'react';
import PropTypes from 'prop-types';

import SearchIcon from '../../../images/search-icon.svg';

const SearchField = ({ value }) => (
  <div className="c-search-field loading">
    <input
      value={value}
      className="search-input"
      placeholder="Search..."
    />
    <SearchIcon width="20" />
  </div>
);

SearchField.displayName = 'SearchField';

SearchField.propTypes = {
  value: PropTypes.string.isRequired
};

export default SearchField;
