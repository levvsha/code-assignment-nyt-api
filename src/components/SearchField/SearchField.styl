.c-search-field {
  align-items: center
  background: rgba(0,0,0,.3)
  border-radius: 10px
  display: flex
  justify-content: space-between
  padding: 0.5em 0.5em 0.5em 1em
  getTransition(all, 0.5s)
  width: 500px
  margin: 0 auto

  &:hover,
  &:focus {
    background: rgba(0,0,0,.4)
  }

  .search-input {
    appearance: none
    background: transparent
    border: 0
    color: inherit
    font: inherit
    outline: 0
  }

  .search-input {
    flex: 1

    &::placeholder {
      color: #fff
    }
  }
}
