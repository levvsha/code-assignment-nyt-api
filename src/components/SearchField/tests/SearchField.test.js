import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SearchField from '../';

describe('<SearchField />', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('SearchField renders correctly', () => {
    const tree = renderer.create(
      <SearchField
        value="foo"
        onChange={() => {}}
        onKeyDown={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('SearchField onChange handler is triggered', () => {
    const onChangeSpy = jest.fn();

    const component = shallow(
      <SearchField
        value="foo"
        onChange={onChangeSpy}
        onKeyDown={() => {}}
      />
    );

    const event = {
      preventDefault: () => {},
      target: { value: 'foot' }
    };

    component.find('input').simulate('change', event);

    expect(onChangeSpy).toHaveBeenCalled();
    expect(onChangeSpy).toHaveBeenCalledWith(event);
    component.unmount();
  });

  it('SearchField onKeyDown handler is triggered', () => {
    const onKeyDownSpy = jest.fn();

    const component = shallow(
      <SearchField
        value="foo"
        onChange={() => {}}
        onKeyDown={onKeyDownSpy}
      />
    );

    const event = {
      preventDefault: () => {},
      keyCode: 123
    };

    component.find('input').simulate('keydown', event);

    expect(onKeyDownSpy).toHaveBeenCalled();
    expect(onKeyDownSpy).toHaveBeenCalledWith(event);
    component.unmount();
  });
});
