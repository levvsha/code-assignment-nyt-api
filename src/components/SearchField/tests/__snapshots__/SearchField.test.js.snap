// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> SearchField renders correctly 1`] = `
<div
  className="c-search-field loading"
>
  <input
    className="search-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search..."
    value="foo"
  />
  <div />
</div>
`;
